lib/graphQl/joiConverter.js

```diff
@@ -18,32 +18,40 @@ joiConverter.simpleAttribute = joiScheme => {
   }
 
   if (type === 'array') {
-    if (joiScheme._inner.items.length === 1) {
-      let joinSubType = joiConverter.simpleAttribute(joiScheme._inner.items[0])
-
-      if (!joinSubType) {
-        throw new Error('Unable to parse Joi type, got ' + JSON.stringify(joiScheme))
-      }
-
-      return new graphQl.GraphQLList(graphQl.GraphQLString)
-    } else {
+    if (joiScheme._inner.items.length !== 1) {
       throw new Error('Joi arrays must contain a single type')
     }
-  }
-
-  const uType = type[0].toUpperCase() + type.substring(1)
-  type = graphQl[`GraphQL${uType}`]
-
-  if (!type) {
-    throw new Error('Unable to parse Joi type, got ' + JSON.stringify(joiScheme))
+    const innerType = joiConverter.simpleAttribute(attribute, joiScheme._inner.items[0], write)
+    type = new graphQl.GraphQLList(innerType)
+  } else if (type === 'object') {
```

Comment:
Yeah, right now putting relationships within nested objects breaks because of this logic. Basically the way we're using Joi to validate relationships involves using object types with additional `_settings` properties. Those happen to fall into the logic I've highlighted and proceeds to iterate through the internal data - it should be using something like the `oneRelationship`/`manyRelationships` types defined at the bottom of this file. Sadly dropping those in doesn't work as GraphQl likes to have duplicate definitions - one for Read and one for Write. Toggling between both leaves me with some.

The GraphQl schema gets parsed at once, after the entire schema has been generated, and the errors it produces are incredibly vague:
```
Error: readarticles_nested.tags field type must be Output Type but got: oneRelationship.
    at invariant (/home/ninj4/repos/jsonapi-server/node_modules/graphql/jsutils/invariant.js:19:11)
    at /home/ninj4/repos/jsonapi-server/node_modules/graphql/type/definition.js:279:29
    at Array.forEach (native)
```
which makes this surprisingly hard to debug. I don't have the time to debug this one right now :( The fix will involve changing how we're iterating through the tree to correctly identify relationships from regular nested objects.